wallet-new/src/Cardano/Wallet/Client.hs

```diff
@@ -149,3 +162,18 @@ data ClientError
     -- ^ This constructor is used when the API client reports an error that
     -- isn't represented in either the 'ServantError' HTTP errors or the
     -- 'WalletError' for API errors.
+    deriving (Show)
+
+-- | General (and naive) equality instance.
+instance Eq ClientError where
+    (==) (ClientWalletError e1) (ClientWalletError e2) = e1 == e2
+    (==) (ClientHttpError   e1) (ClientHttpError   e2) = e1 == e2
+    (==) (UnknownError      _ ) (UnknownError      _ ) = True
```

Comment:
it just occured to me that you could write `UnknownError e1 == UnknownError e2 = displayException e1 == displayException e2` but that's a little dastardly